### DISCUSSION TITLE
[Idea] Replace obsolete jquery.simplePagination.js in yarn by another modern library in npm

### DISCUSSION THREAD

**Discussion by leemyongpakvn** [PrestaShop/PrestaShop#31624] [Ideas]
### The problem

Module productcomments is using an [obsolete pagination](https://github.com/PrestaShop/productcomments/blob/v5.0.2/views/js/jquery.simplePagination.js) hosted on [yarn](https://github.com/PrestaShop/productcomments/tree/v5.0.2#update-dependency).

### The solution

Replace current pagination library by another modern one hosted on npm like [this](https://www.npmjs.com/package/paginationjs) 

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [x] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by dennispw** [comment]
In all honesty it doesn't even need a library or plugin. It would probably work just fine with [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) as long as there is some sort of "`page`" variable to use.

**Reply by leemyongpakvn** [reply]
We can use fetch instead of jquery get. But a library is needed for Numbers and Navigation of the Page listing. I'd prefer Replace by similar than Rewrite from scratch :)

**Comment by kpodemski** [comment]
I like the idea and agree that it's not worth reinventing the wheel and writing it from scratch. This library you mentioned is quite heavy tho 🤔

**Reply by kpodemski** [reply]
@Oksydan, any thoughts on this subject Igor?

**Reply by leemyongpakvn** [reply]
Yes, similar feature but triple heavier is a big problem :confused:  I will find other solution.

**Reply by Oksydan** [reply]
Hi,
What about not using this kind of library at all?
We can move that template back to smarty files and build pagination inside template files. 
No libraries needed just simple fetch and replace content.

**Reply by leemyongpakvn** [reply]
Without 3rd pagination library, we can generate page number buttons and prev/next buttons in template files, most of current CSS and JS of module can be reused but some new JS code is needed to handle those buttons click.
By the way, I worry that once this module is migrated to webpack / vue, the custom pagination will be replaced by vue pagination or prestashop-ui-kit pagination.

**Reply by Oksydan** [reply]
Why we should migracje this module to vue? This module is very simple and not interactive enough to require such technology like vue. We could remove most of js code from module. We don't need to generate customer comment via js. We could easily move it to template and even add hook for example displayCusyomerCommentBottom. You could open window to extend module behavior via third party module. Customer service response to customer comment for example.

**Reply by leemyongpakvn** [reply]
It is just my worry. We should not to migrate this module to vue, but to webpack is very possible.
I will try to replace the jquery.simplePagination.js library by fetch API and some custom JS code.
Other contributors can try to remove most of js code from module later ;)

**Reply by kpodemski** [reply]
@leemyongpakvn great 👍🏻🎉

There's no plan to migrate this module to Vue 👍🏻

**Reply by leemyongpakvn** [reply]
@kpodemski How can I _Create issue from this Discussion_ ? This is not a Bug report nor a Security vulnerability

**Comment by leemyongpakvn** [comment]
Solved by https://github.com/PrestaShop/productcomments/pull/157
